File: new_post.py
#!/usr/bin/env -S uv run --script
#
# /// script
# requires-python = ">=3.6"
# dependencies = []
# ///

from datetime import date
import sys


today: date = date.today()
title_raw: list[str] = sys.argv[1:]

if not title_raw:
    raise RuntimeError("Title cannot be empty")

title_spaced: str = " ".join(title_raw)
title_dashed: str = "-".join(title_raw)
today_dashed: str = today.strftime("%Y-%m-%d")

post = f"""---
layout: post
title: {title_spaced}
created: {today_dashed}
---

"""

# only open the file for writing if it does not exist,
# with "x" mode
with open(f"{today_dashed}-{title_dashed}.md", "x", encoding="utf-8") as f:
    f.write(post)

File: new_post.exs
#!/usr/bin/env elixir

Mix.install(tzdata: "~> 1.1")
Calendar.put_time_zone_database(Tzdata.TimeZoneDatabase)

today = DateTime.now!("America/Chicago")
title_raw = System.argv()

if Enum.empty?(title_raw) do
  raise "Title cannot be empty"
end

title_spaced = Enum.join(title_raw, " ")
title_dashed = Enum.join(title_raw, "-")
today_dashed = Calendar.strftime(today, "%Y-%m-%d")

post = [
  "---",
  "layout: post",
  "title: #{title_spaced}",
  "created: #{today_dashed}",
  "---",
  "\n\n"
]

post = Enum.intersperse(post, "\n")

File.write!("#{today_dashed}-#{title_dashed}.md", post, [:exclusive])
